src/GraphQL.Harness/CountFieldMiddleware.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using GraphQL.Instrumentation;
+using GraphQL.StarWars;
+using GraphQL.Types;
+using Microsoft.AspNetCore.Http;
+
+namespace Example
+{
+    /// <summary>
+    /// Example of Field Middleware with dependencies. When configured as singleton in DI
+    /// counts fields execution for the entire application lifetime. When configured as
+    /// scoped counts how many fields were executed when processing the single request.
```

Comment:
```suggestion
    /// scoped (with a scoped schema), counts how many fields were executed when processing the single request.
```
Or something similar